### PR TITLE
plugins/auth-backend: added basic SAML auth

### PR DESCRIPTION
**Pull request by Rugvip**
About to start up some SAML experiments, but adding an empty backend first

### CODE REVIEW

**Review by freben — APPROVED** [review]
Not connected in the backend itself yet tho

**Comment by Rugvip** [comment]
Yeah needs to run on a separate origin anyway

**Comment by Rugvip** [comment]
Needs a re-review 😅

File: plugins/auth-backend/src/providers/PassportStrategyHelper.ts [hunk 1]
```diff
@@ -51,11 +51,12 @@ export const executeFrameHandlerStrategy = async (
     strategy.error = (error: Error) => {
       reject(new Error(`Authentication failed, ${error}`));
     };
-    strategy.redirect = () => {
-      reject(new Error('Unexpected redirect'));
+    strategy.redirect = (url: string) => {
```
Comment (soapraj) [inline plugins/auth-backend/src/providers/PassportStrategyHelper.ts:54]: this needs to be reverted back?

File: plugins/auth-backend/src/providers/factories.ts [hunk 2]
```diff
@@ -14,37 +14,37 @@
...
-import {
-  AuthProviderFactories,
-  AuthProviderRouteHandlers,
-  AuthProviderConfig,
-} from './types';
-import { GoogleAuthProvider } from './google';
-import { GithubAuthProvider } from './github';
-import { OAuthProvider } from './OAuthProvider';
+import Router from 'express-promise-router';
+import { createGithubProvider } from './github/provider';
```
Comment (soapraj) [inline plugins/auth-backend/src/providers/factories.ts]: change import to github/index instead

**Review by soapraj — APPROVED** [review]

**Review by freben — APPROVED** [review]

**Pull request merged**